File: Backend/src/main/kotlin/demo/controller/DashboardController.kt
package demo.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
@RequestMapping("/")
class DashboardController {

    private val random = Random()

    @GetMapping
    fun getRoot(): String {
        return "Welcome to the backend API! The server is running."
    }

    @GetMapping("/dashboard")
    fun getDashboard(): List<Int> {
        val testList = mutableListOf<Int>()
        var last = random.nextInt(100)
        for(i in 0 .. 4) {
            val next = if(i % 2 == 0) {
                last + ((random.nextInt(40) + 10) * last / 100)
            } else{
                last - ((random.nextInt(40) + 10) * last / 100)
            }
            last = next
            testList += next
        }

        return testList
    }
}

File: Backend/Controllers/DashboardController.cs
using Microsoft.AspNetCore.Mvc;

namespace Backend.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class DashboardController : ControllerBase
    {
        [HttpGet]
        public List<int> Get()
        {
            List<int> testList = new List<int>();
            Random rand = new Random();
            int last = rand.Next(50, 100);

            for (int i = 0; i < 5; i++)
            {
                int next;
                if (i % 2 == 0)
                {
                    next = last + (rand.Next(10, 50) * last / 100);
                }
                else
                {
                    next = last - (rand.Next(10, 50) * last / 100);
                }
                last = next;
                testList.Add(next);
            }
            return testList;
        }
    }

}

